kong/pdk/tracing.lua

```diff
@@ -158,7 +158,7 @@ local function create_span(tracer, options)
       or generate_trace_id()
   local sampled = span.parent and span.parent.should_sample
       or options.should_sample
-      or tracer and tracer.sampler(trace_id)
+      or not span.parent and tracer and tracer.sampler(trace_id) -- TODO: is this ok?
```

Comment:
this means that if the parent span had disabled sampling, we set sampled to false in this span too